learning_resources/content_summarizer.py

```diff
@@ -0,0 +1,229 @@
+import logging
+from typing import Annotated, Optional
+
+from django.db import transaction
+from django.db.models import Q
+from langchain_community.chat_models import ChatLiteLLM
+from typing_extensions import TypedDict
+
+from learning_resources.models import (
+    ContentFile,
+    ContentSummarizerConfiguration,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class Flashcard(TypedDict):
+    """Flashcard structure model"""
+
+    question: Annotated[str, ..., "The question for the flashcard"]
+    answer: Annotated[str, ..., "The answer for the flashcard"]
+
+
+class FlashcardsResponse(TypedDict):
+    """Flashcards response model"""
+
+    flashcards: Annotated[list[Flashcard], ..., "A list of generated flashcards"]
+
+
+class ContentSummarizer:
+    """
+    A service class to generate summaries and flashcards for the ContentFiles.
+    """
+
+    def _can_process_content_file(self, content_file: ContentFile) -> bool:
+        """Check if a content file can be processed
+        Args:
+            - content_file (ContentFile): Content file to check
+        Returns:
+            - bool: Whether the content file can be processed
+        """
+
+        if not content_file.content:
+            # Summarizer needs content to process, if there is no content then skip
+            return False
+
+        summarizer_config = getattr(
+            content_file.run.learning_resource.platform, "summarizer_config", None
```

Comment:
Currently in the system a contentfile can be associated with either a run or a learning resource directly. (We probably wont encounter issues since we are only processing particular contentfiles that should always have a run) it might be safer to do something like:
```
platform = content_file.run.learning_resource.platform if content_file.run else content_file.learning_resource.platform
summarizer_config = getattr(platform, "summarizer_config", None)
```